kafka/consumer.go

```diff
@@ -1062,3 +1062,10 @@ func (c *Consumer) handleRebalanceEvent(rkev *C.rd_kafka_event_t) (retval Event)
 
 	return nil
 }
+
+// MemberID returns this client's broker-assigned group member id.
+func(c *Consumer) MemberID() string {
+	cMemberId := C.rd_kafka_memberid(c.handle.rk)
+	defer C.rd_kafka_mem_free(c.handle.rk, unsafe.Pointer(cMemberId))
```

Comment:
I think this needs a null check before freeing - documentation says 

```
 * @returns An allocated string containing the current broker-assigned group
 *          member id, or NULL if not available.
 *          The application must free the string with \p free() or
 *          rd_kafka_mem_free()
```

And bad things tend to happen if you call free(NULL) ;)

Reply:
Thanks for the review @KJTsanaktsidis 🙇‍♀️ 
Based on[ Is it good practice to free a NULL pointer in C?](https://stackoverflow.com/questions/6084218/is-it-good-practice-to-free-a-null-pointer-in-c) which quoted the C standard, 7.20.3.2/2 from[ ISO-IEC 9899](http://www.open-std.org/JTC1/SC22/wg14/www/docs/n1124.pdf) 

> If ptr is a null pointer, no action occurs.

It seems that it's okay to not check NULL, do you think it applies to our case?

Reply:
Huh, TIL! Looks like rd_kafka_mem_free just calls free()

https://github.com/edenhill/librdkafka/blob/6a1595f5e28a3627e2e0efc1bf28ea3dcdb094c7/src/rdkafka.c#L4883
https://github.com/edenhill/librdkafka/blob/063a9ae7a65cebdf1cc128da9815c05f91a2a996/src/rd.h#L138

so this should be OK.